src/manage_sql.c

```diff
@@ -57038,7 +57038,7 @@ check_openvasd_result_exists (report_t report, task_t task,
   else if (res->port > 0)
     {
       char buf[6];
-      snprintf (buf, sizeof(buf) , "%d", res->port);
+      snprintf (buf, sizeof(buf) , "%s", res->port);
```

Comment:
This needs a better solution, because the code around it is still expecting port to be an int. For example, `buf` is only 6 chars long.

I expect the whole if-elseif-else block can be replaced with `port_str = res->port ?: ""` but @jjnicola will have a better idea.

Note that currently `port_str` is leaking, and `port_str = ""` should have been `port_str = g_strdup ("")`.